plugins/kubernetes/app/assets/javascripts/kubernetes/factories/build_factory.js

```diff
@@ -0,0 +1,16 @@
+samson.factory('buildFactory', function() {
```

Comment:
This seems like a very Java-like pattern. I don't think Angularjs factories are meant to be used like Java factories.